src/equations/shallow_water_multilayer_1d.jl

```diff
@@ -421,18 +517,20 @@ end
 
 # Convert primitive to conservative variables
 @inline function Trixi.prim2cons(prim, equations::ShallowWaterMultiLayerEquations1D)
-    H = prim[1:nlayers(equations)]
-    v = prim[(nlayers(equations) + 1):(2 * nlayers(equations))]
+    # To extract the interface height and velocity we reuse the waterheight and momentum functions 
+    # from the conservative variables.
+    H = waterheight(prim, equations)    # For primitive variables this extracts the interface height
+    v = momentum(prim, equations)       # For primitive variables this extracts the velocity
     b = prim[end]
 
     # Calculate waterheight
     h = MVector{nlayers(equations), real(equations)}(undef)
     for i in eachlayer(equations)
         if i < nlayers(equations)
-            setindex!(h, H[i] - H[i + 1], i)
+            h[i] = H[i] - H[i + 1]
         else
             # The lowest layer is measured from the bottom topography
-            setindex!(h, H[i] - b, i)
+            h[i] = H[i] - b
```

Comment:
Why remove the set index?

Reply:
This is only done for better readability. Both option work and do not allocate.

Reply:
Okay, then the current version is better. It is more explicit what is being done.